src/Nox.Lib/Middlewares/RelatedEndpointsMiddleware.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Microsoft.AspNetCore.Http;
+using Nox.Lib;
+using Nox.Solution;
+
+namespace Nox.Middlewares;
+
+internal class RelatedEndpointsMiddleware
+{
+    private readonly PathString _apiPrefix;
+
+    private readonly RequestDelegate _next;
+
+    private readonly IEnumerable<string> _entitiesPluralNames;
+
+    private readonly IEnumerable<string> _navigationNames;
+
+    public RelatedEndpointsMiddleware(RequestDelegate next, NoxSolution solution)
+    {
+        _next = next;
+
+        _apiPrefix = solution.Presentation.ApiConfiguration.ApiRoutePrefix;
+
+        if (solution.Domain is null)
+        {
+            _entitiesPluralNames = Array.Empty<string>();
+            _navigationNames = Array.Empty<string>();
+        }
+        else
+        {
+            _entitiesPluralNames = solution.Domain.Entities.Select(e => e.PluralName).ToList();
+            _navigationNames = solution.Domain.Entities
+                .SelectMany(e => e.Relationships.Select(r => e.GetNavigationPropertyName(r)))
+                .Distinct()
+                .ToList();
+        }
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        var requestPath = context.Request.Path;
+
+        if (!requestPath.HasValue)
```

Comment:
to remove this to if/branch, I would suggest on the middleware registration to do something like:
`builder.UseWhen(context => context.Request.Path.StartsWithSegments("solution.Presentation.ApiConfiguration.ApiRoutePrefix")`

This can be applied to both middleware

Reply:
updated